### PR TITLE
13746-Add missing transfers to child, add documentid

### PR DESCRIPTION
**Pull request by ChasBoucher**
*Issue #:* /bcgov/entity#13746

*Description of changes:*

- Add missing Transfers 
- Add "Completed" ux status for complted transfers
- Add documentid to provide uniqueness for table - follows existing ppr logic.
- Fix deletion/editing of draft

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the PPR license (Apache 2.0).

### CODE REVIEW

**Review by cameron-eyds — APPROVED** [review]
Looks fine to me!
Have you tested against existing Registrations Table (PPR)?
Also, what is the documentId used for specifically?

File: ppr-ui/src/composables/mhrRegistration/useNewMhrRegistration.ts [hunk 1]
```diff
@@ -208,21 +208,34 @@ export const useNewMhrRegistration = () => {
     MhRegistrationSummaryIF[] {
     const sortedDraftFilings = orderBy(mhrDrafts, ['createDateTime'], ['desc'])
 
-    const registerNewUnit = mhrHistory.filter(registrations =>
```
Comment (dimak1) [inline ppr-ui/src/composables/mhrRegistration/useNewMhrRegistration.ts:211]: Considering this composable is used in a few places, is it safe to update the logic here? Won't it break anything pre-existing?

**Review by dimak1 — COMMENTED** [review]
@ChasBoucher so far looks good, just have one comment.

File: ppr-ui/src/composables/mhrRegistration/useNewMhrRegistration.ts (hunk 1 above)
Comment (ChasBoucher) [reply ppr-ui/src/composables/mhrRegistration/useNewMhrRegistration.ts:211]: Glad you made me check. We're good. This is the only place that MhrHistory is being used at present.  I checked the api call and although I added the optional parameter, it is only being used here as well.

**Comment by ChasBoucher** [comment]
DocumentId is the key used in the table.  It needs to be unique or a warning is thrown. 
Ppr Registrations are ok, but I just notice a few issues with the Staff role that I want to fix...

**Comment by cameron-eyds** [comment]
> DocumentId is the key used in the table. It needs to be unique or a warning is thrown. Ppr Registrations are ok, but I just notice a few issues with the Staff role that I want to fix...

Sounds good thanks, so `documentId` is generated and returned by the backend i'm assuming. 
There is probably a business rule around that! Helpful in providing a unique ID for us then :) 
Thanks for the looking into it.

**Comment by cameron-eyds** [comment]
@ChasBoucher Sorry Charles, 1 more item noticed after the DRAFT refactors. 
When you resume a DRAFT using the EDIT button, i no longer see it pulling on the DRAFT itself but the original Registration again. 
Does this PR also resolve that?

**Comment by cameron-eyds** [comment]
@ChasBoucher Sorry Charles, 1 more item noticed after the DRAFT refactors. 
When you resume a DRAFT using the EDIT button, i no longer see it pulling on the DRAFT itself but the original Registration again. 
Does this PR also resolve that?

**Comment by ChasBoucher** [comment]
> @ChasBoucher Sorry Charles, 1 more item noticed after the DRAFT refactors. When you resume a DRAFT using the EDIT button, i no longer see it pulling on the DRAFT itself but the original Registration again. Does this PR also resolve that?

Just tested and this pr does address the issue with editing of drafts as well.

**Comment by cameron-eyds** [comment]
> Collaborator

Excellent, thank you 👍

**Review by cameron-eyds — APPROVED** [review]

**Pull request merged**